spirv_msl.cpp

```diff
@@ -1588,6 +1589,25 @@ void CompilerMSL::preprocess_op_codes()
 	                          (need_subpass_input_ms && !msl_options.use_framebuffer_fetch_subpasses))))
 		needs_sample_id = true;
 
+	// OpKill is removed by the parser, so we need to identify those by inspecting
+	// blocks.
+	ir.for_each_typed_id<SPIRBlock>([&preproc](uint32_t, SPIRBlock &block) {
+		if (block.terminator == SPIRBlock::Kill)
+			// FIXME: This doesn't distinguish between OpKill and OpTerminateInvocation!
```

Comment:
OpKill == OpTerminateInvocation, so this is fine I think. Kill is just looser where it gives a little leeway to support demote because broken apps/drivers, but that was never the intention.

Reply:
One day, though, I'll need to do this differently, in order to properly support `SPV_KHR_terminate_invocation`. Basically, we'll need to set a boolean, like now, only now we immediately return from all functions when we determine an `OpTerminateInvocation` was executed.